resources/js/Pages/Home.vue

```diff
@@ -121,62 +116,41 @@
         errors : { [ key : string ] : string }
     }
 
-    export default defineComponent({
-        components: {
-          CdxDialog,
-          CdxButton,
-          CdxIcon,
-          CdxMessage,
-          ItemIdSearchTextarea,
-          InertiaHead
-        },
-        setup() {
-            const store = useStore();
-            const textareaInputValue = ref(store.lastSearchedIds);
-            
-            return {
-                cdxIconDie,
-                cdxIconInfo,
-                textareaInputValue
-            };
-        },
-        methods: {
-            send(): void {
-                (this.$refs.textarea as InstanceType<typeof ItemIdSearchTextarea>).validate();
-
-                if((this.$refs.textarea as InstanceType<typeof ItemIdSearchTextarea>).validationError) {
-                    return;
-                }
-                const store = useStore();
-                store.saveSearchedIds( this.textareaInputValue );
-                this.$inertia.get( '/results', 
-                    { ids: (this.$refs.textarea as InstanceType<typeof ItemIdSearchTextarea>).serializeInput() }
-                );
-            },
-            showRandom(): void {
-                this.$inertia.get( '/random' );
-            },
-        },
-        computed: {
-            serversideValidationError() {
-                const errors = this.$page.props.errors as ErrorMessages;
-                return errors && Object.keys(errors).length > 0;
-            },
-            unexpectedError() {
-                const flashMessages = this.$page.props.flash as FlashMessages;
-                return (flashMessages.errors && flashMessages.errors.unexpected);
-            },
-            // spread to combine with local computed props
-            // only mapping 'loading' and not 'lastSearchedIds' because computed
-            //properties are not available when data is processed in vue's lifecycle
-            ...mapState(useStore, ['loading']),
-        },
-        data(): HomeState {
-            return {
-                validationError: null,
-                faqDialog: false
-            }
+    const textarea = ref<InstanceType<typeof ItemIdSearchTextarea> | null>(null);
+    const store = useStore();
+    const page = usePage();
+    const textareaInputValue: Ref<string> = ref(store.lastSearchedIds);
+    const faqDialog = ref(false);
+
+    function send(): void{
+        (textarea as InstanceType<typeof ItemIdSearchTextarea>).value.validate();
+
+        if((textarea as InstanceType<typeof ItemIdSearchTextarea>).value.validationError) {
+            return;
         }
+        const store = useStore();
```

Comment:
We already have store in scope, we don't need to use it again.

Reply:
oops, yes, that was a copy/paste bloop. good catch. thx